### PR TITLE
feat: Added Signup and Login API View

### PR DESCRIPTION
**Pull request by Jawayria**

### CODE REVIEW

File: estore_project/users/views.py [hunk 1]
```diff
@@ -1,45 +1,60 @@
...
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        user = User.objects.get(username=self.request.data["username"])
+
+        refresh = RefreshToken.for_user(user)
+        res = {
```
Comment (mraarif) [inline estore_project/users/views.py:27]: why don't we look for a package that would take care of creating tokens for us

**Comment by mraarif** [comment]
could you also fix the CI

File: estore_project/users/views.py (hunk 1 above)
Comment (Jawayria) [reply estore_project/users/views.py:27]: I am using 'Refresh token' here from rest_framework_simplejwt.tokens to create token. Do you think we should use some other package?

Comment (mraarif) [reply estore_project/users/views.py:27]: `simplejwt` provides a class based view `TokenObtainPairView` that takes care of doing the authentication, creating and returning the toekns

Comment (Jawayria) [reply estore_project/users/views.py:27]: Okay

File: estore_project/users/urls.py [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+from rest_framework_simplejwt.views import TokenRefreshView
 
-from .views import Login, Signup
+from .views import LoginView, Signup
 
 app_name = "users"
 urlpatterns = [
     path("signup/", view=Signup.as_view(), name="signup"),
-    path("login/", view=Login.as_view(), name="login"),
+    path("login/", LoginView.as_view(), name="login"),
```
Comment (mraarif) [inline estore_project/users/urls.py:9]: why not do  `path("login/", TokenObtainPairView.as_view(), name="login")` and remove the `LoginView`

Comment (Jawayria) [reply estore_project/users/urls.py:9]: I need to get user_id in response as well along with the access and refresh tokens so I wrote my own serializer and mentioned that serializer in LoginView.

**Review by mraarif — APPROVED** [review]

**Pull request merged**